config/validate_server.go

```diff
@@ -1,22 +1,35 @@
 package config
 
 import (
+	"time"
+
 	validation "github.com/go-ozzo/ozzo-validation"
 	"github.com/go-ozzo/ozzo-validation/is"
 )
 
 // Validate checks the configuration for specific rules
 func (s *ServerConfig) Validate() error {
-
 	// Set defaults
 	if s.IdleTimeout.String() == "0s" {
-		s.IdleTimeout = DefaultHTTPRequestIdleTimeout
+		duration, err := time.ParseDuration(ServerIdleTimeoutDefault)
+		if err != nil {
+			return err
+		}
+		s.IdleTimeout = duration
```

Comment:
I don't like this idea here.
This is method is called `Validate`, therefore if the s.IdleTImeout is not expected number (for example 0) then we should just return error, instead of setting the default. The other code is responsible for defaults.
The same applies to lines below